lib/TemplatedPathPlugin.js

```diff
@@ -102,13 +102,24 @@ const replacePathVariables = (path, data, assetInfo) => {
 	// [ext] - .js
 	if (data.filename) {
 		if (typeof data.filename === "string") {
-			const idx = data.filename.indexOf("?");
+			const queryIdx = data.filename.indexOf("?");
 
 			let file, query;
 
-			if (idx >= 0) {
-				file = data.filename.substr(0, idx);
-				query = data.filename.substr(idx);
+			if (queryIdx >= 0) {
+				file = data.filename.substr(0, queryIdx);
+
+				const maybeQuery = data.filename.substr(queryIdx);
+
+				if (maybeQuery.length > 1) {
```

Comment:
Do not add the `?` character for requests like `cosnt file = import('./image.png?')`